server.go

```diff
@@ -10,11 +10,22 @@ import (
 	"google.golang.org/grpc"
 )
 
-// PreregisterServices takes a gRPC server and pre-initializes all counters to 0.
-// This allows for easier monitoring in Prometheus (no missing metrics), and should be called *after* all services have
-// been registered with the server.
-func Register(server *grpc.Server) {
-	serviceInfo := server.GetServiceInfo()
+// Server defines the interface that is required for grpc_prometheus to Register.
+type Server interface {
+	GetServiceInfo() map[string]grpc.ServiceInfo
+}
+
+// Register takes a gRPC server and pre-initializes all counters to 0.
+// This allows for easier monitoring in Prometheus (no missing metrics), and
+// should be called *after* all services have been registered with the server.
+func Register(server Server) {
```

Comment:
I'm trying to fix (in a backwards compatible way) the RPC registration path to be more generalized (and thus more generally useful), as well as trying to avoid contorting my non-`*grpc.Server` code to pretend that it's that object. The current code base for registering RPCs (and the interface only change for `Regsiter`) is saying:
> in order to register RPCs give me an object that I can call an arbitrary method on to return a description of those RPCs

What I believe it should say instead is:
> in order to register RPCs give me a description of those RPCs

That stops this library from randomly picking a method name of `GetServiceInfo` that all consumers must implement and instead spells out exactly what it needs: `map[string]grpc.ServiceInfo`, i.e. "a description of those RPCs".

With `RegisterServiceInfo` it fulfills that more generalized registration functionality. Any code that followed the server-side usage example can simply do:
```
grpc_prometheus.RegisterServiceInfo(myServer.GetServiceInfo())
```
And literally any other non-`*grpc.Server` code can do:
```
var infos map[string]grpc.ServiceInfo
// collect infos here from whatever is servicing them, however it is appropriate to.
grpc_prometheus.RegisterServiceInfo(infos)
```

If we're willing to change to an interface to make more generally useful, why not also address the original overly constrained path so that people can migrate away from `Register`?